PayPal/src/main/java/com/braintreepayments/api/PayPalCheckoutRequest.java

```diff
@@ -0,0 +1,246 @@
+package com.braintreepayments.api;
+
+import android.text.TextUtils;
+
+import androidx.annotation.StringDef;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Represents the parameters that are needed to start the PayPal Checkout flow
+ */
+public class PayPalCheckoutRequest extends PayPalRequest {
+
+    @Retention(RetentionPolicy.SOURCE)
+    @StringDef({PayPalCheckoutRequest.INTENT_ORDER, PayPalCheckoutRequest.INTENT_SALE, PayPalCheckoutRequest.INTENT_AUTHORIZE})
+    @interface PayPalPaymentIntent {}
+
+    /**
+     * Payment intent to create an order
+     */
+    public static final String INTENT_ORDER = "order";
+
+    /**
+     * Payment intent for immediate payment
+     */
+    public static final String INTENT_SALE = "sale";
+
+    /**
+     * Payment intent to authorize a payment for capture later
+     */
+    public static final String INTENT_AUTHORIZE = "authorize";
+
+    @Retention(RetentionPolicy.SOURCE)
+    @StringDef({PayPalCheckoutRequest.USER_ACTION_DEFAULT, PayPalCheckoutRequest.USER_ACTION_COMMIT})
+    @interface PayPalPaymentUserAction {}
```

Comment:
Does this need a docstring? Explaining what these two enum values mean?

Reply:
I've added a general docstring for the `PayPalPaymentUserAction` and `PayPalPaymentIntent` types.